crates/services/sync/src/import.rs

```diff
@@ -214,17 +218,27 @@ where
         .then({
             let state = state.clone();
             let executor = executor.clone();
-            move |block| {
+            let p2p = p2p.clone();
+            move |res| {
+                let p2p = p2p.clone();
                 let state = state.clone();
                 let executor = executor.clone();
                 async move {
-                    // Short circuit on error.
-                    let block = match block {
-                        Ok(b) => b,
-                        Err(e) => return Err(e),
-                    };
-
-                    execute_and_commit(executor.as_ref(), &state, block).await
+                    let (peer_id, block) = res?;
+
+                    let res = execute_and_commit(executor.as_ref(), &state, block).await;
+                    match &res {
+                        Ok(_) => {
+                            let _ = p2p.report_peer(peer_id.clone(), PeerReportReason::SuccessfulBlockImport)
+                                .await
+                                .map_err(|e| tracing::error!("Failed to report successful block import for peer {:?}: {:?}", peer_id, e));
+                        },
+                        Err(_) => {
+                            // If this fails, then it means that consensus has approved a block that is invalid.
```

Comment:
We should at least log that this is a serious error instead of silently failing. We could even introduce some kind of error backoff or shutdown the sync service altogether so we don't keep importing the same bad block over and over every 5s.

Reply:
I'll log it for now. Any concept of sending a shutdown signal or something would fall out-of-scope for this story. Happy to create a new issue.

Reply:
I think this should be captured in a new issue and completed separately.

Reply:
https://github.com/FuelLabs/fuel-core/issues/1341